### DISCUSSION TITLE
Any mobile client which supports tags/labels

### DISCUSSION THREAD

**Discussion by dawidry** [FreshRSS/FreshRSS#5196] [Q&A] [answered]
Hi,

I setup Freshrss on my own Synology recently. It works perfectly. 
I love built in feature where I can add my tags/labels to articles.

And here is question: Do you know any mobile client (iOS) which support this feature? 
I've tried:
NetNewsWire - I love and use this client for years but it treats labels as folders (empty folders) 
Fluent Reader - it ignore labels (skip it even if are defined in FreshRSS)
ReadKit - it states in features list: "Organize subscriptions / articles into folder / tags" But I cannot figure out how to use "tags"

**Comment by Alkarex** [comment, marked as answer]
Hello,
The only mobile app that I know has full support for tags/labels is the (old) Android app: [News+](https://github.com/noinnion/newsplus/blob/master/apk/NewsPlus_202.apk) with [Google Reader extension](https://github.com/noinnion/newsplus/blob/master/apk/GoogleReaderCloneExtension_101.apk)

Only mobile apps using the Google Reader API have the hability to support labels, since it is not a concept supported by the Fever API.

See the column *Labels* in the following table (updates welcome):
https://github.com/FreshRSS/FreshRSS/blob/edge/README.md#apis--native-apps

It would be great to motivate a few contributors to send some pull requests adding this feature to some of the open source apps of this table.

**Reply by dawidry** [reply]
Thanks for reply and clear explanation that is greader API specific. 
Actually I sent message on NetNewsWire github about this feature :) We will see. That would be really cool to have this feature supported in mobile/desktop application. Let's hope.

**Comment by xiaopaul** [comment]
@dawidry Hi Davidry, may I know how do you setup it on Synology? I have the same environment, but don't know how to use it. Thank you.

**Reply by dawidry** [reply]
Hi,
I guess I used this guideline:
https://mariushosting.com/how-to-install-freshrss-on-your-synology-nas/
Then (to have access from outside of my network e.g. mobile phone) I set up subdomain with Reverse Proxy.

**Reply by xiaopaul** [reply]
Thank you @dawidry 
I also followed the guidance in the article your suggested before. It works well. But Cannot let mobile app connect it. As I cannot get the API.

As your mentioned, you can have the API once you do set up the Reverse proxy? How do you make it? Only in Synology setup， or 3rd party application? Thank you for further help.

**Reply by dawidry** [reply]
OK, so I assume you have up and running FreshRSS. You have access to it locally.
Next is to point any domain/subdomain to this service. So the question is are you familiar with Reverse Proxy and are you using it? I'm using build in Synology Reverse Proxy. You'll find it in Login Portal -> Advanced -> Reverse Proxy. And below there is my example for FreshRSS. But if you didn't use Reverse Proxy before you have to follow one of many guideline e.g. on you youtube how it works. It's nothing much more than I'm showing below but you also need to do port forwarding on your router. So this post doesn't cover how to setup Reverse Proxy from start to end rather shows how to setup FreshRSS API endpoints. But no 3rd party apps are needed. Just Synology build-in features + FreshRSS.

<img width="1152" alt="freshrss-reverse" src="https://github.com/FreshRSS/FreshRSS/assets/13475936/c7de7176-c093-4ca5-9ccc-9f98af894286">

Then in FressRSS configuration panel (when you're logged in as administrator) go to Profile. You'll see something like shown below.

<img width="588" alt="freshrss-api" src="https://github.com/FreshRSS/FreshRSS/assets/13475936/b025622d-4eed-43ed-adbb-519759c36635">

1. At the bottom create API password first.
2. Then click link below password input (https://......./api). This will lead you to the page as shown below.

<img width="364" alt="freshrss-api-address" src="https://github.com/FreshRSS/FreshRSS/assets/13475936/e313e971-4fd5-4f42-8854-f156cd408337">

As you can see I have marked them as PASS. But initially (at least in my case) I had Warning Sign here. That's was because by default my URL was build with local ip address e.g 192.168....+port
To fix it you have to change base url. To do so go (with Filestation) to FreshRSS home folder. If you followed by guideline installation which I provided in previous post this should be:
`docker -> freshrss -> www -> freshrss -> data`

- open config.php in text editor
- change property 'base_url' => .....................    <--- provide here URL which you configured with Reverse Proxy here
- save it

and now refresh page "FreshRSS API endpoints". Now both endpoints should have status PASS.
So now to connect to FreshRSS from mobile (conjunction with API password + username) use one of this URLs (depending which mobile client you use).

**Reply by xiaopaul** [reply]
@dawidry 
Hi Dawidry, thank you very much. It works now. Your guidance are very helpful.

Thank you again.

**Reply by xiaopaul** [reply]
Every step is on the right place.  Thank you.

**Reply by dawidry** [reply]
I'm glad that it is helpful.
Enjoy FreshRSS on your own Synology. It's really works flawlessly. After 2 months I'm super happy.
